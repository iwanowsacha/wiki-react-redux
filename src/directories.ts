import path from 'path';
import * as fs from 'fs-extra';
import { DirectoriesList } from './types';

export const DIRECTORIES: { [key: string]: string } = {
  articles: path.join(__dirname, 'articles'),
  lists: path.join(__dirname, 'lists'),
};

const createDirectoryIfNotExists = async (directoryPath: string) => {
  const exists = await fs.pathExists(directoryPath);
  if (!exists) fs.mkdir(directoryPath);
};

const prepareDirectories = async () => {
  const promises = Object.values(DIRECTORIES).map((directoryPath: string) =>
    createDirectoryIfNotExists(directoryPath)
  );
  await Promise.all(promises).catch(console.log);
};

export const loadDocuments = async () => {
  const directoriesList: DirectoriesList = {};

  try {
    await prepareDirectories();

    Object.entries(DIRECTORIES).forEach(([key, directory]) => {
      directoriesList[key] = fs.readdirSync(directory);
    });
    // for (const directory of Object.values(DIRECTORIES)) {
    //   directoriesList[
    //     directory.slice(directory.lastIndexOf('/') + 1)
    //   ] = await fs.readdir(directory);
    // }
  } catch (err) {
    console.log(err);
  }
  console.log(directoriesList);
  return directoriesList;
};
