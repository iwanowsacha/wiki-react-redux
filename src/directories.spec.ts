import * as fs from 'fs-extra';
import { createDirectoryIfNotExists } from './directories';

jest.mock('fs-extra');

describe('directories', () => {
  describe('createDirectoryIfNotExists', () => {
    beforeAll(async () => {
      console.log('before');
      // @ts-ignore
      fs.mkdir.mockClear();

      await createDirectoryIfNotExists('some/test/path');

      // @ts-ignore
      fs.pathExists.mockReturnValue(true);

      await createDirectoryIfNotExists('some/test/path');
    });

    it('should match snapshot of calls (mkdir)', () => {
      // @ts-ignore
      expect(fs.mkdir.mock.calls).toMatchSnapshot();
    });

    it('should match snapshot of calls (pathExists)', () => {
      // @ts-ignore
      expect(fs.pathExists.mock.calls).toMatchSnapshot();
    });

    it('(mkdir) should have been called with...', () => {
      expect(fs.mkdir).toHaveBeenCalledWith('some/test/path', {
        recursive: true,
      });
    });

    it('(pathExists) should have been called with...', () => {
      expect(fs.pathExists).toHaveBeenCalledWith('some/test/path');
    });

    it('should have been called 1 time (mkdir)', () => {
      expect(fs.mkdir).toHaveBeenCalledTimes(1);
    });

    it('should have been called 2 times (pathExists)', () => {
      expect(fs.pathExists).toHaveBeenCalledTimes(2);
    });
  });
});
