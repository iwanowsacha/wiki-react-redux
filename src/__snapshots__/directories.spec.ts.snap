// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directories createDirectoryIfNotExists should match snapshot of calls (mkdir) 1`] = `
Array [
  Array [
    "some/test/path",
    Object {
      "recursive": true,
    },
  ],
]
`;

exports[`directories createDirectoryIfNotExists should match snapshot of calls (pathExists) 1`] = `
Array [
  Array [
    "some/test/path",
  ],
  Array [
    "some/test/path",
  ],
]
`;
